pymc3/variational/approximations.py

```diff
@@ -322,7 +353,12 @@ def elbo(self, samples=1, pi=1):
         samples = tt.as_tensor(samples)
         pi = tt.as_tensor(pi)
         posterior = self.posterior(samples)
-        elbo = self.log_p_D(posterior) - pi * self.KL_q_p_W(posterior)
+        elbo = (self.log_p_D(posterior)
+                - self.log_q_W_local(posterior)
+                - pi * self.log_q_W_global(posterior)
+                + self.log_p_W_local(posterior)
+                + pi * self.log_p_W_global(posterior))
```

Comment:
@ferrine Sorry, my explanation was wrong. log_q_W_local should be weighted with (M * pi) where M is the number of mini-batches. According to Eq. (8) in the paper, the ELBO term relevant to samples should be scaled M times of  the global term. And prior KL terms are scaled proportional to pi, while leaving likelihood terms without scaling.

``` python
elbo = (self.log_p_D(posterior)
            - M * pi * self.log_q_W_local(posterior)
            - pi * self.log_q_W_global(posterior)
            + M * pi * self.log_p_W_local(posterior)
            + pi * self.log_p_W_global(posterior))
```

We can avoid hardcoding for weighting as follows:

``` python
elbo = (self.log_p_D(posterior)
            - pi_local * self.log_q_W_local(posterior)
            - pi_global * self.log_q_W_global(posterior)
            + pi_local * self.log_p_W_local(posterior)
            + pi_global * self.log_p_W_global(posterior))
```
In the later case, pi_local/global should be appropriately set by the user, or in a child class of BaseApproximation.